packages/logger/src/Logger.ts

```diff
@@ -847,11 +849,17 @@ class Logger extends Utility implements LoggerInterface {
           >);
 
     this.console[consoleMethod](
-      JSON.stringify(
-        log.getAttributes(),
-        this.getJsonReplacer(),
-        this.logIndentation
-      )
+      typeof log === 'string' ? log : this.formatLog(log)
```

Comment:
If I understand this change correctly, we're bypassing the creation of an object, including all the attributes when passing a string message only.

If this is the case I don't think we would ever make this change, since the main value proposition of this logger is to emit JSON structured logs with keys, context, and all the other features.

Reply:
I probably should have explained this change more thoroughly.
When saving logs into the buffer instead of emiting them to stdout, we cannot save the original object in the buffer because this object might be mutated later on in the Lambda function:

```ts
const user = { name: 'John' };
logger.debug('User', { user } );
user.name = 'Jane';
``` 

So I added a new function `formatLog` which is identical to `printLog` but doesn't emit to stdout:

https://github.com/aws-powertools/powertools-lambda-typescript/blob/e6d0db9978682fc104bf74ff0ddb5fbba63e798f/packages/logger/src/Logger.ts#L856-L864 

I use this function to serialise the log object into JSON and save it to the buffer: 

https://github.com/aws-powertools/powertools-lambda-typescript/blob/e6d0db9978682fc104bf74ff0ddb5fbba63e798f/packages/logger/src/Logger.ts#L907-L913

Then when we flush the log, since we already have a JSON string we only need to print it to stdout:

https://github.com/aws-powertools/powertools-lambda-typescript/blob/e6d0db9978682fc104bf74ff0ddb5fbba63e798f/packages/logger/src/Logger.ts#L884-L887

Reply:
I see - makes sense, I missed this.

Thank you for explaining it.